### DISCUSSION TITLE
collection viewについて

### DISCUSSION THREAD

**Discussion by ghost** [lit-iphone-course/lit-iphone-course#30] [Question-original]
(1) 実現したいこと
・スワイプしたら次の月、前の次に切り替わるようにカレンダーのcollection viewを設定したい
・itemboxviewcontrollerのcollection viewが表示されない
(2) 実現するために自分が試した内容とその結果
(3) 調べた内容や資料
https://i-app-tec.com/ios/collectionview.html
https://naoya-ono.com/swift/swift-collection-view-basic/
https://rara-world.com/swift-collection-view-programming/
https://ios-docs.dev/collectionview/
https://qiita.com/sasao3/items/581b3f1bb9da0d4c4d89
(4) 原因だと思われそうな箇所(ソースコードをコピペして貼る)

(5) GitHubのリンク
https://github.com/j-fuuuu/calender.git

**Comment by txxxxc** [comment]
結論から言うと、UICollectionViewを使ってカレンダーを表現するのは期限的にも結構大変かもしれないです！(表現したいデザイン的にはUICollectionViewが良さそうなのは分かります！)
なので個人的には別の方法を推したいです！それを踏まえて2つ方法を共有します！

###  UIDatePickerを使う
UIDatePickerを使うと結構かんたんにカレンダーから日付選択をする機能が実装できます

参考↓
[iOS14のUIDatePickerのデザインをカスタマイズする – Ideal Reality](https://ideal-reality.com/programing/ios14-datepicker-design/)

### ライブラリを使う
FSCalendarというカレンダーを簡単に扱うことができるライブラリがあるので、それを使うと結構簡単にできるかもです！
ただ日本語の情報が少なくて、英語で書かれたドキュメントを読む必要が出てくる気がするので、速さを求めるなら上の方法がおすすめです！(個人的には一旦上で実装して時間があったらこの方法を試して見るくらいのほうがいいのかもと思ってます...!)

[[ライブラリ] FSCalendarを使用する(1) storyboardへの貼り付けと使用準備 - Swift Life](http://swift.hiros-dot.net/?p=65)


参考にしてみてくださいー！

**Comment by yukiny0811** [comment]
https://github.com/yuruyuru-learning/calendar-test
slackで伝えたけどもし余裕があったらこれ見てやってみてねー！
